Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -394,11 +394,10 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
             descriptionText.characters.count > 0
             else { return }
         
-        var position = description.position
-
-        // if no position specified, draw on default position
-        if position == nil
-        {
+        let position: CGPoint
+        if let descriptionPosition = description.position {
+            position = descriptionPosition
+        } else {
             let frame = self.bounds
```

Comment:
It's generally good practice to short a variable call  (i.e. `let width = self.frame.size.width`), but in this case it is unnecessary and wrong. Unnecessary because you don't need to explicitly state `self`, and wrong because in Cocoa `frame` and `bounds` mean two different things. `frame` should have never been here in the first place.

Reply:
done